lua/kommentary/init.lua

```diff
@@ -5,49 +5,98 @@ This module handles the initialization of the plugin.
 ]]
 local kommentary = require("kommentary.kommentary")
 local config = require("kommentary.config")
+local util = require("kommentary.util")
 local M = {}
+local context = util.enum({"line", "visual", "motion", "init"})
+local modes = config.get_modes()
 
 --[[--
-Toggle comment on current line, selection or motion.
+Function to be called by mappings.
+This acts as a middle-man, when called it will figure out the context of the
+call, so if it was called on a single line, or from a selection/motion, and
+also the range of the operation, so on which line to start and end.
+It will then call another function which takes care of calling the functions
+in kommentary.kommentary, which then actually comments out the range.
+You can overload this function with another function as an argument, which
+will overwrite the function that gets called at the end. The arguments
+passed to the second function are: line_number_start, line_number_end, context
 @tparam string ... Optional string indicating the mode to choose, 'single_line'
 	will operate on the current line, 'visual' will use the current visual
 	selection, otherwise it will assume a motion (Arguments will be
 	automatically passed by operatorfunc).
+@tparam ?string|function ... Optional function to call instead of the normal
+toggle_comment at the end of this function
 @treturn ?string|nil If called without arguments, it sets itself as operatorfunc
 	and returns 'g@' to be used in an expression mapping, otherwise it will
 	comment out and not return anything.
 ]]
-function M.toggle_comment(...)
+function M.go(...)
     local args = {...}
+    -- The 3 possible contexts from which this function can be called
```

Comment:
```suggestion
    -- The first argument passed to this function represents one of 3 possible contexts in which this function can be called.
```
Very unclear comment.